serverless/src/getProposals/__tests__/handler.test.js

```diff
@@ -0,0 +1,268 @@
+import { mockClient } from 'aws-sdk-client-mock'
+import { GetObjectCommand, S3Client } from '@aws-sdk/client-s3'
+import { sdkStreamMixin } from '@smithy/util-stream'
+
+import getProposals from '../handler'
+import * as s3ListObjects from '../../utils/s3ListObjects'
+
+const s3ClientMock = mockClient(S3Client)
+
+beforeEach(() => {
+  s3ClientMock.reset()
+  process.env.COLLECTION_PROPOSALS_BUCKET_NAME = 'test-bucket'
+})
+
+beforeAll(() => {
+  vi.spyOn(console, 'log').mockImplementation(() => {})
+})
+
+afterAll(() => {
+  vi.restoreAllMocks()
+})
+
+describe('getProposals', () => {
+  describe('when the proposals are found', () => {
+    test('returns the proposals from s3', async () => {
+      const bucketContentsMock = vi.spyOn(s3ListObjects, 's3ListObjects').mockResolvedValue([
+        {
+          Key: 'proposals/123',
+          LastModified: '2023-01-01T00:00:00.000Z'
+        },
+        {
+          Key: 'proposals/456',
+          LastModified: '2023-01-02T00:00:00.000Z'
+        }
+      ])
+
+      s3ClientMock
+        .on(GetObjectCommand, { Key: 'proposals/123' })
+        .resolves({
+          Body: sdkStreamMixin(new Blob([JSON.stringify({
+            providerId: 'MMT-1',
+            shortName: 'Proposal 1',
+            entryTitle: 'Entry 1',
+            proposalStatus: 'SUBMITTED'
+          })]))
+        })
+        .on(GetObjectCommand, { Key: 'proposals/456' })
+        .resolves({
+          Body: sdkStreamMixin(new Blob([JSON.stringify({
+            providerId: 'MMT-2',
+            shortName: 'Proposal 2',
+            entryTitle: 'Entry 2',
+            proposalStatus: 'APPROVED'
+          })]))
+        })
+
+      const event = { queryStringParameters: {} }
+      const response = await getProposals(event)
+
+      expect(response.statusCode).toBe(200)
+
+      const parsedBody = JSON.parse(response.body)
+      expect(parsedBody).toHaveLength(2)
+      expect(parsedBody[0]).toEqual(expect.objectContaining({
+        id: '123',
+        providerId: 'MMT-1',
+        shortName: 'Proposal 1',
+        entryTitle: 'Entry 1',
+        proposalStatus: 'SUBMITTED'
+      }))
+
+      expect(parsedBody[1]).toEqual(expect.objectContaining({
+        id: '456',
+        providerId: 'MMT-2',
+        shortName: 'Proposal 2',
+        entryTitle: 'Entry 2',
+        proposalStatus: 'APPROVED'
+      }))
+
+      expect(bucketContentsMock).toHaveBeenCalledTimes(1)
+      expect(bucketContentsMock).toHaveBeenCalledWith(
```

Comment:
```suggestion
      expect(bucketContentsMock).toHaveBeenCalledWith(
```
uh I think expect.anything() is kind of odd here